### PR TITLE
A feature to disable / enable single wireguard client configs

### PR DESCRIPTION
**Pull request by shelleycat485**
Addresses #1111
Code like the delete, but comment out lines for the user in configs, wg0.conf instead of delete.
-dis or disable
-ena or enable
-ena -v will give a list of clients disabled
-dis -v will give a list of clients disabled

Tested OK on AWS Ubuntu 20.04 and AWS Ubuntu 18.04 AMD64. Have to do an apt-get update and apt-get upgrade before the install works. Tested creating client, connecting, disabling, enabling, multiple client create, client delete, -l and -c functions. For some reason installation has to be root on the AWS machines.
Also tested Rpi3+ Buster: getting Warning: The unit file, source configuration file or drop-ins of wg-quick@wg0.service changed on disk. Run 'systemctl daemon-reload' to reload units.

### CODE REVIEW

**Comment by orazioedoardo** [comment]
I've not tested this yet, however I would change the command line behavior:
- `pivpn on` to enable
- `pivpn off` to disable
- `pivpn list` shows disabled clients as well, instead of a separate command

**Comment by shelleycat485** [comment]
Will do

**Comment by shelleycat485** [comment]
Tested OK on pi4.

File: scripts/wireguard/pivpn.sh [hunk 1]
```diff
@@ -80,6 +92,8 @@ showHelp(){
     echo ":::  -l,  list             List all clients"
     echo ":::  -qr, qrcode           Show the qrcode of a client for use with the mobile app"
     echo ":::  -r,  remove           Remove a client"
+    echo ":::  -off,  off            Disable a user"
+    echo ":::  -on ,  on             Enable a user"
```
Comment (orazioedoardo) [inline scripts/wireguard/pivpn.sh:96]: Add space to the lines above and below these so the commas and the text line up vertically

File: scripts/wireguard/bash-completion [hunk 2]
```diff
@@ -4,8 +4,8 @@ _pivpn()
     COMPREPLY=()
     cur="${COMP_WORDS[COMP_CWORD]}"
     prev="${COMP_WORDS[COMP_CWORD-1]}"
-    dashopts="-a -c -d -l -qr -r -h -u -up -bk"
-    opts="add clients debug list qrcode remove help uninstall update backup"
+    dashopts="-a -c -d -l -qr -r -h -u -up -bk -dis -ena"
+    opts="add clients debug list qrcode remove help uninstall update backup disable enable"
```
Comment (orazioedoardo) [inline scripts/wireguard/bash-completion:8]: Old completion options.

File: scripts/wireguard/disableCONF.sh [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+	*)
+            CLIENTS_TO_CHANGE+=("$1")
+            ;;
+    esac
+    shift
+done
+
+cd /etc/wireguard
+if [ ! -s configs/clients.txt ]; then
+    echo "::: There are no clients to remove"
```
Comment (orazioedoardo) [inline scripts/wireguard/disableCONF.sh:49]: change instead of remove

**Comment by orazioedoardo** [comment]
When a cient is disabled, `pivpn -c` won't work as it reads `/etc/wireguard/configs/clients.txt` which gives no clue that it is indeed disabled. Perhaps the file should include the status of the client as the last number of the line "1" or 0" for example, which could be queried by `pivpn on`, `pivpn off`, `pivpn list`.

Example `pivpn list`:

```
::: Clients Summary :::
Client      Public key                                        Creation date                Status
user        PAcspHIYKQtWzZtHX4cLJvZhDEW9yuCFKBWA2XYvpBg=      05 Dec 2020, 13:50, GMT      Disabled
```

```
$ pivpn -c
::: Connected Clients List :::
/opt/pivpn/wireguard/clientSTAT.sh: line 49: [: : integer expression expected
numfmt: invalid number: ‘’
numfmt: invalid number: ‘’
Name        Remote IP      Virtual IP      Bytes Received      Bytes Sent      Last Seen
user                                                                           (not yet)
```

I would add parenthesis and a space to make the wg0.conf easier to read:
```
#[disabled] ### begin user ###
#[disabled] [Peer]
#[disabled] PublicKey = PAcspHIYKQtWzZtHX4cLJvZhDEW9yuCFKBWA2XYvpBg=
#[disabled] PresharedKey = tJvmkZlbErA4LztNj8/NJb0fwbwVmwG4UkQ2q/tTAL0=
#[disabled] AllowedIPs = 10.6.0.2/32
#[disabled] ### end user ###
```

File: scripts/wireguard/disableCONF.sh (hunk 3 above)
Comment (shelleycat485) [reply scripts/wireguard/disableCONF.sh:49]: will do

**Comment by shelleycat485** [comment]
Will address this

**Comment by shelleycat485** [comment]
I've got a problem with the logic now -the pivpn -c command uses the wireguard config to get the information about the clients, but the  disabled ones are not in that config file - they are not known to the wg0 service.   So I don't see how to get the disabled clients listed on the connected clients command.  I could do the same as the pivpn -l command when I add the disabled ones to the bottom of the list.

**Comment by shelleycat485** [comment]
all  comments above addressed, doing testing

File: scripts/wireguard/bash-completion (hunk 2 above)
Comment (shelleycat485) [reply scripts/wireguard/bash-completion]: done

**Comment by orazioedoardo** [comment]
Still getting an error on `pivpn -c` if all clients are disabled:

```
pi@raspberrypi:~ $ pivpn -c
::: Connected Clients List :::
/opt/pivpn/wireguard/clientSTAT.sh: line 49: [: : integer expression expected
numfmt: invalid number: ‘’
numfmt: invalid number: ‘’
Name       Remote IP      Virtual IP      Bytes Received      Bytes Sent      Last Seen
test1
test2                                                                         (not yet)
::: Disabled clients :::
[disabled]   test1 
[disabled]   test2
```

> I've got a problem with the logic now -the pivpn -c command uses the wireguard config to get the information about the clients, but the disabled ones are not in that config file

I would have added the disabled status in the client.txt file https://github.com/pivpn/pivpn/pull/1186#issuecomment-739257286

**Comment by shelleycat485** [comment]
Changed to all clients disabled worked, tested on rpi3

**Pull request merged**